lib/loader-steps.js

```diff
@@ -0,0 +1,510 @@
+        if (!config.bagtainer.scan.enable) {
+            fulfill(passon);
+        } else if (!clam) {
+            fulfill(passon);
+        } else {
+            debug('Scanning %s for viruses at path %s', passon.id, passon.bagpath);
+            clam.scan_dir(passon.bagpath, (error, good, bad) => {
+                if (error) {
+                    debug(error);
+                    reject(error);
+                } else {
+                    debug('Virus scan completed and had %s good and >> %s << bad files', good.length, bad.length);
+                    if (bad.length > 0) {
+                        debug('Virus found, deleting directory  %s', passon.bagpath);
+
+                        let badfiles = bad.join('\n\t');
+                        debug('Found bad files in:\n\t%s', badfiles);
+
+                        exec('rm -r ' + passon.bagpath, (error, stdout, stderr) => {
+                            if (error || stderr) {
+                                debug(error, stderr, stdout);
+                                debug('Error deleting compendium with virus. File deleted by virus checker? %s)',
+                                    clam.settings.remove_infected);
+                            } else {
+                                debug('Deleted directory %s', passon.bagpath);
+                            }
+
+                            if (emailTransporter) {
+                                let mail = {
+                                    from: config.bagtainer.scan.email.sender, // sender address 
+                                    to: config.bagtainer.scan.email.receivers,
+                                    subject: '[o2r platform] a virus was detected during upload',
+                                    text: 'A virus was detected in a compendium uploaded by user ' + passon.user +
+                                    ' in these files:\n\n' + JSON.stringify(bad)
+                                };
+
+                                emailTransporter.sendMail(mail, function (error, info) {
+                                    if (error) {
+                                        debug('Problem sending notification email: %s', error.message);
+                                    }
+                                    debug('Email sent: %s\n%s', info.response, JSON.stringify(mail));
+                                });
+                            }
+
+                            let msg = 'Virus scan found infected file(s) in directory'
+                            let err = new Error(msg);
+                            err.status = 422;
+                            err.msg = msg;
+                            reject(err);
+                        });
+                    } else {
+                        debug('No viruses found in %s', passon.id);
+                        fulfill(passon);
+                    }
+                }
+            });
+        }
+    });
+}
+
+function extractMetadata(passon) {
```

Comment:
Please update methods related to metadata from `muncher/master`, see changes in https://github.com/o2r-project/o2r-muncher/pull/43